File: Backend/UAM-LIFT/src/main/java/ni/edu/uam/UAM_LIFT/models/Usuario.java
package ni.edu.uam.UAM_LIFT.models;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.SQLDelete;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "usuarios")
@SQLDelete(sql = "UPDATE usuarios SET estado = false WHERE id = ?")
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(length = 100, unique = true, nullable = false)
    private String cif;
    @Column(unique = true, nullable = false, length = 50)
    private String nombreUsuario;
    @Column(nullable = false, length = 50)
    private String nombre;
    @Column(nullable = false, length = 50)
    private String apellido;
    @Column(nullable = false, length = 50, unique = true)
    private String correo;
    @Column(nullable = false, length = 100, unique = true)
    private String contraseña;
    @Column(nullable = false, length = 100)
    private String imagenUrl;
    @Column(nullable = false)
    private boolean estado = true;
    @Column(nullable = false)
    private boolean correoVerificado = false;
    @Column(nullable = false)
    private int numeroViajes = 0;

}

File: Uam Lift/app/src/main/java/ni/edu/uam/uamlift/models/Usuario.kt
package ni.edu.uam.uamlift.models

import com.google.gson.annotations.SerializedName

data class Usuario(
    @SerializedName("id")
    val id: Long? = null,
    @SerializedName("cif")
    val cif: String? = null,
    @SerializedName("nombreUsuario")
    val nombreUsuario: String? = null,
    @SerializedName("nombre")
    val nombre: String? = null,
    @SerializedName("apellido")
    val apellido: String? = null,
    @SerializedName("correo")
    val correo: String? = null,
    @SerializedName("contraseña")
    val contrasenia: String? = null,
    @SerializedName("imagenUrl")
    val imagenUrl: String? = null,
    @SerializedName("estado")
    val estado: Boolean = true,
    @SerializedName("correoVerificado")
    val correoVerificado: Boolean = false
)
